src/hooks/useProposalFilters.js

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useWallet } from '../providers/Wallet'
+import {
+  filterArgsMapping,
+  NULL_FILTER_STATE,
+  RANKING_FILTER_TOP,
+  RANKING_ITEMS,
+  STATUS_FILTER_OPEN,
+  STATUS_ITEMS,
+  SUPPORT_ITEMS,
+  TYPE_ITEMS,
+} from '../utils/filter-utils'
+
+const INITIAL_PROPOSAL_COUNT = 10
+const PROPOSAL_COUNT_STEP = 5
+
+// Status and Type filters will be used as subgraph query args.
+// Support will be filtered locally as there is not an easy way to achieve this only by querying the subgraph.
+export default function useProposalFilters() {
+  const { account } = useWallet()
+  const [proposalCount, setProposalCount] = useState(INITIAL_PROPOSAL_COUNT)
+  const [rankingFilter, setRankingFilter] = useState(RANKING_FILTER_TOP)
+  const [statusFilter, setStatusFilter] = useState(STATUS_FILTER_OPEN)
+  const [supportFilter, setSupportFilter] = useState(NULL_FILTER_STATE)
+  const [typeFilter, setTypeFilter] = useState(NULL_FILTER_STATE)
+
+  const handlePoposalCountIncrease = useCallback(index => {
+    setProposalCount(count => count + PROPOSAL_COUNT_STEP)
+  }, [])
+  const handleRankingFilterChange = useCallback(index => {
+    setRankingFilter(index)
+  }, [])
+  const handleStatusFilterChange = useCallback(
+    index => setStatusFilter(index || NULL_FILTER_STATE),
+    []
+  )
+  const handleSupportFilterChange = useCallback(
+    index => setSupportFilter(index || NULL_FILTER_STATE),
+    []
+  )
+  const handleTypeFilterChange = useCallback(
+    index => setTypeFilter(index || NULL_FILTER_STATE),
+    []
+  )
+
+  const handleClearFilters = useCallback(() => {
+    setStatusFilter(STATUS_FILTER_OPEN)
+    setSupportFilter(NULL_FILTER_STATE)
+    setTypeFilter(NULL_FILTER_STATE)
+  }, [])
+
+  useEffect(() => {
+    if (!account) {
+      setSupportFilter(NULL_FILTER_STATE)
+    }
+  }, [account])
+
+  const isActive =
```

Comment:
this active is to know if the proposal is still active? if so, shouldn't we just check for the status?

Reply:
Ah no, this is to indicate that the filters are active (meaning they are not the default ones). Do you think the naming is confusing?

Reply:
ah maybe yes, we can rename it to filtersActive or something like that maybe